File: laptop/janus/janus/vertex.py
from edge import Edge

global vertex_id
vertex_id = 0


class Vertex:
    def __init__(self):
        global vertex_id
        self.id = vertex_id
        vertex_id += 1
        self.edges = []
        self.label = None
        self.cluster_id = 0

    def add_edge(self, vertex_creating_edge_to, weight):
        for edge in self.edges:
            if edge.get_other(self).id == vertex_creating_edge_to:
                return
        new_edge = Edge(self, vertex_creating_edge_to, weight)
        self.edges.append(new_edge)
        vertex_creating_edge_to.edges.append(new_edge)

File: laptop/janus/backend/c/vertex.c
/*
 * vertex.c
 *
 *  Created on: Apr 22, 2018
 *      Author: matt
 */
#include <stdlib.h>
#include "vertex.h"
static int id = 0;
struct _vertex * initVertex(
		struct _vertex * self,
		char * label,
		float weight,
		int clusterId)
{
	self = (Vertex)malloc(sizeof(struct _vertex));
	self->id = id++;
	self->clusterId = clusterId;
	self->weight = weight;
	self->getEdgeWeightTotal = getEdgeWeightTotal;
	if (label != NULL)
	{
		int stringSize = strlen(label);
		self->label = (char*)calloc(stringSize + 1, sizeof(char));
		strncpy(self->label, label, stringSize);
	}
	return self;
}

float getEdgeWeightTotal (struct _vertex * self)
{
	Enode travler = self->edges;
	float total = 0;
	while (travler != NULL)
	{
		total += travler->edge->weight;
		travler = travler->next;
	}
	return total;
}
